### PR TITLE
Authenticate on CI

### LINKED ISSUE #25: Hitting GitHub API rate limit on CI
We're getting 403s from a GitHub public on the mac GitHub actions build for the open source repo. Probably someone else on GitHub actions hitting the rate limit and we're sharing their IP.

This is a shame, it means our build can be influenced by other peeps. We should move to an authenticated client on CI to ensure we don't hit the limit

It would be good for authentication to be optional as to not require it for development.

It would be good to use the Embark bot as the owner of the API token so that it continues to work after the developer has left. We will need to determine the appropriate permissions scope and then log into that account to make the token https://github.com/embark-bot

### PR DESCRIPTION
**Pull request by lpil**
### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

Use GitHub API token to avoid hitting rate limit on CI.

The CI build shows it working, tada

### Related Issues

Closes #25

### CODE REVIEW

File: src/github.rs [hunk 1]
```diff
@@ -9,6 +9,74 @@ use lazy_static::lazy_static;
...
+
+    async fn api_get_response(&self, url: &str) -> eyre::Result<reqwest::Response> {
+        let request = reqwest::Client::new()
+            .get(url)
+            .header("accept", "application/vnd.github.v3+json")
+            .header("user-agent", "embark-oss");
+        let request = match &self.github_api_token {
+            Some(token) => request.header("authorization", format!("token {}", token)),
+            _ => request,
+        };
```
Comment (lpil) [inline src/github.rs:70]: This is the main change here

File: src/github.rs [hunk 2]
```diff
@@ -9,6 +9,74 @@ use lazy_static::lazy_static;
...
+        Ok(collection)
+    }
+
+    async fn api_get_response(&self, url: &str) -> eyre::Result<reqwest::Response> {
+        let request = reqwest::Client::new()
+            .get(url)
+            .header("accept", "application/vnd.github.v3+json")
+            .header("user-agent", "embark-oss");
+        let request = match &self.github_api_token {
+            Some(token) => request.header("authorization", format!("token {}", token)),
```
Comment (soniasingla) [inline src/github.rs:68]: here, `format!("token {}", token)),` shouldn't be used with `&`, we need to pass value by reference!

**Review by soniasingla — COMMENTED** [review]
LGTM 🎉

**Comment by lpil** [comment]
Thanks Sonia!

**Pull request merged**